node_helper.js

```diff
@@ -798,9 +799,14 @@ module.exports = NodeHelper.create({
 
         if (result.variants.length > 0) {
           result.variants.sort((a, b) => { return a.width * a.height - b.width * b.height; });
-          result.url = result.variants[result.variants.length - 1].url;
-          self.flickrDataCache.set(p.id, result, config.flickrDataCacheTime);
-          images.push(result);
+          let selection = _.findLast(result.variants, function (variant) {
+            return (variant.width <= config.maxWidth) && (variant.height <= config.maxHeight);
+          });
+          if (selection !== undefined) {
+            result.url = selection.url;
+            self.flickrDataCache.set(p.id, result, config.flickrDataCacheTime);
+            images.push(result);
+          }
```

Comment:
I'd rather not add another dependency for such a simple operation, could:

```suggestion
          let selection = result.variants.reduce((prev, variant) => {
            return ((variant.width <= config.maxWidth) && (variant.height <= config.maxHeight)) ? variant : prev;
          }, undefined);
          if (selection !== undefined) {
            result.url = selection.url;
            self.flickrDataCache.set(p.id, result, config.flickrDataCacheTime);
            images.push(result);
          }
```

Reply:
Thanks. That works. Will rebase it into the branch to keep the commit history tidy for such a small diff.